package/MDAnalysis/core/topology.py

```diff
@@ -17,13 +17,58 @@
 Topology object --- :mod:`MDAnalysis.core.topology'
 ===================================================================
 
+
+"""
+
+"""
+# second docstring == multiline comment
+
+TODO Notes:
+  Could make downshift tables lazily built! This would
+    a) Make these not get built when not used
+    b) Optimise moving multiple atoms between residues as only built once afterwards
+
+  Could optimise moves by only updating the two parent tables rather than rebuilding everything!
+
+
 """
+from six.moves import zip
 import numpy as np
 
 from ..lib.mdamath import one_to_many_pointers
 from .topologyattrs import Atomindices, Resindices, Segindices
 
 
+def make_downshift_arrays(upshift):
+    """From an upwards translation table, create the opposite direction
+
+    Turns a many to one mapping to a one to many mapping
+
+    Eg :: 
+      AR = np.array([0, 1, 0, 2, 2, 0, 2])
+    
+      make_downshift_arrays(AR)
+      array([array([0, 2, 5]), array([1]), array([3, 4, 6])], dtype=object)
+
+    So entry 0 informs that items 0, 2 & 5 all belong to group 0.
+
+    Returns
+    -------
+    Numpy array of numpy arrays (dtype object)
+    Length : 
+    """
+    order = np.argsort(upshift)
+
+    upshift_sorted = upshift[order]
+    borders = [None] + list(np.nonzero(np.diff(upshift_sorted))[0] + 1) + [None]
+
+    # returns an array of arrays
+    # Add None to end of array to force it to be of type Object
+    # Without this, a rectangular array gets squashed into a single array
+    return np.array([np.array(order[x:y], copy=True, dtype=np.int)
```

Comment:
So this is really ugly but what I wanted to achieve was an array of arrays.  If the input arrays are all the same size, it ends up as a 2 dimensional array.  To get around this, I add a `None` entry at the end to force it to be an array of dtype `object`.

It needs to be an array of arrays so that individual arrays can be resized for residue moves, if it starts 2 dimensional, it can't later be resized fully.

If there's an obvious way to do this I've missed I'd be happy to be corrected